File: Problemset/0038-Count and Say/Two Pointers.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s = "1"
        for i in range(n - 1):
            tmp = ""
            i = 0
            for j in range(len(s)):
                if s[i] != s[j]:
                    tmp += str(j - i) + s[i]
                    i = j
            tmp += str(len(s) - i) + s[i]
            s = tmp
        return s

File: Problemset/0038-Count and Say/Two Pointers.rb
# @param {Integer} n
# @return {String}
def count_and_say(n)
    s = "1"

    for _ in 1...n
        tmp = ""
        i = 0

        for j in 0...s.length
            if s[i] != s[j]
                tmp += (j - i).to_s + s[i]
                i = j
            end
        end

        tmp += (s.length - i).to_s + s[i]
        s = tmp
    end

    return s
end
